service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -528,14 +543,177 @@ public Optional<LoadTableResponse> loadTableIfStale(
+          (IcebergCatalog.TableMetadataProvider) ic.newTableOps(tableIdentifier);
+      StreamingLoadTableResponse res =
+          ops.processLatestMetadata(
+              (fileIO, tableEntity) -> {
+                return new StreamingLoadTableResponse(
+                    tableEntity.getMetadataLocation(),
+                    fileIO.newInputFile(tableEntity.getMetadataLocation()).newStream(),
+                    Map.of(),
+                    List.of());
+              },
+              () -> null);
+      if (res == null) {
+        throw new NoSuchTableException("Table does not exist: %s", tableIdentifier.toString());
+      }
+      return Optional.of(res);
+    }
     return Optional.of(CatalogHandlers.loadTable(baseCatalog, tableIdentifier));
   }
 
-  public LoadTableResponse loadTableWithAccessDelegation(
+  public static class StreamingLoadTableResponse implements RESTResponse {
+    private String metadataLocation;
+    private InputStream metadata;
+    private Map<String, String> config;
+    private List<Credential> credentials;
+
+    public StreamingLoadTableResponse() {}
+
+    public StreamingLoadTableResponse(
+        String metadataLocation,
+        InputStream metadata,
+        Map<String, String> config,
+        List<Credential> credentials) {
+      this.metadataLocation = metadataLocation;
+      this.metadata = metadata;
+      this.config = config;
+      this.credentials = credentials;
+    }
+
+    @Override
+    public void validate() {
+      Preconditions.checkNotNull(this.metadata, "Invalid metadata: null");
+    }
+
+    @JsonProperty("metadata-location")
+    public String metadataLocation() {
+      return this.metadataLocation;
+    }
+
+    @JsonProperty("metadata")
+    @JsonSerialize(using = InputStreamSerializer.class)
+    public InputStream tableMetadata() {
+      return this.metadata;
+    }
+
+    @JsonProperty("config")
+    public Map<String, String> config() {
+      return this.config != null ? this.config : Map.of();
+    }
+
+    @JsonProperty("storage-credentials")
+    public List<Credential> credentials() {
+      return this.credentials != null ? this.credentials : List.of();
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("metadataLocation", this.metadataLocation)
+          .add("metadata", this.metadata)
+          .add("config", this.config)
+          .toString();
+    }
+  }
+
+  public static class InputStreamSerializer extends RawSerializer<InputStream> {
+    public InputStreamSerializer() {
+      super(InputStream.class);
+    }
+
+    @Override
+    public void serialize(
+        InputStream inputStream, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+        throws IOException {
+      LoggerFactory.getLogger(getClass()).warn("Using custom serializer for input stream");
+      byte[] buffer = new byte[4096];
+      try (inputStream) {
```

Comment:
Nit: I would rather model the field as an `InputFile`, then create a serializer for `InputFile`. This would make it clear when the input stream is created and closed.
```suggestion
      try (InputStream inputStream = inputFile.newStream()) {
```